### PR TITLE
docs(dev-server): Add documentation for `serveIndex` option

### PR DESCRIPTION
**Pull request by EslamHiko**
add documentation for `serveIndex` option in this PR 

https://github.com/webpack/webpack-dev-server/pull/1752

### CODE REVIEW

File: src/content/configuration/dev-server.md [hunk 1]
```diff
@@ -1108,6 +1109,22 @@ Usage via the CLI
 webpack-dev-server --quiet
 ```
 
+## `devServer.serveIndex`
+
+`boolean`
+
+This option Enables/Disables [`serveIndex`](https://github.com/expressjs/serve-index) middleware. It's enabled by default.
```
Comment (EugeneHlushko) [inline src/content/configuration/dev-server.md]:
```suggestion
Tells dev-server to use [`serveIndex`](https://github.com/expressjs/serve-index) middleware when enabled.
```

Comment (EslamHiko) [reply src/content/configuration/dev-server.md]: @EugeneHlushko Done 😄

**Comment by EslamHiko** [comment]
@EugeneHlushko some checks are failed, can I do something about them?

**Comment by EugeneHlushko** [comment]
We are testing netlify deploy for previewing PRs, no need to do anything

File: src/content/configuration/dev-server.md (hunk 1 above)
Comment (EugeneHlushko) [reply src/content/configuration/dev-server.md]: thank you, By the way, you can directly commit proposed changes via github interface button: "Commit suggestion"

**Comment by EugeneHlushko** [comment]
Thanks!

**Comment by EugeneHlushko** [comment]
in case anyone interested, here is our first preview: https://5cba05c10f05471a05b57258--webpackjsorg-pr-previews.netlify.com/

**Comment by EslamHiko** [comment]
@EugeneHlushko You're Welcome 😄

**Pull request merged**